libs/shared/contentful/src/lib/contentful.stripe-mapper.spec.ts

```diff
@@ -0,0 +1,337 @@
+      const metadata = {};
+      const mapper = new PlanMapper(
+        defaultCommonContent,
+        defaultPurchaseDetails,
+        metadata
+      );
+      const expected = undefined;
+      const actual = mapper.getStripeForMetadataKey(
+        StripeMetadataKeys.NewsletterSlug
+      );
+      expect(actual).toBe(expected);
+    });
+  });
+
+  describe('mapStripeToContentful', () => {
+    it('successfully maps contentful and stripe data', () => {
+      const defaultMapper = new PlanMapper(
+        defaultCommonContent,
+        defaultPurchaseDetails,
+        stripeMetadata
+      );
+      const result = defaultMapper.mapStripeToContentful();
+      expect(result.errorFields.length).toBe(1);
+      expect(result.errorFields[0]).toBe('emailIconURL');
+      expect(result.metadata.webIconURL).toBe(defaultPurchaseDetails.webIcon);
+    });
+  });
+});
+
+const getPlanById = (plans: Stripe.Plan[], planId: string) =>
+  plans.find((plan) => plan.id === planId);
+
+const getMessagesByPlanId = (messages: string[], planId: string) => {
+  const message = messages.find((message) => {
+    const messagePlanId = message.split(' - ')?.[0];
+    return messagePlanId === planId;
+  });
+  return message?.split(' - ')[1];
+};
+
+describe('ContentfulStripeMapper', () => {
+  const validMetadata = MetadataFactory({
+    webIconURL: 'https://example.com/webIconURL',
+  });
+  const plans = [
+    PlanFactory({
+      id: 'plan1',
+      product: ProductFactory({
+        metadata: validMetadata,
+      }),
+    }),
+    PlanFactory({
+      id: 'plan2',
+      product: ProductFactory({
+        metadata: validMetadata,
+      }),
+    }),
+    PlanFactory({
+      id: 'plan3',
+      product: 'product_id',
+      metadata: validMetadata,
+    }),
+    PlanFactory({
+      id: 'plan4',
+      product: ProductFactory({
+        metadata: validMetadata,
+      }),
+      metadata: MetadataFactory({
+        webIconURL: 'https://plan.override/emailIcon',
+      }),
+    }),
+    PlanFactory({
+      id: 'plan5',
+    }),
+    PlanFactory({
+      id: 'plan6',
+    }),
+  ];
+  const mockContentfulConfigUtil = {
+    transformedPurchaseWithCommonContentForPlanId: (
+      planId: string
+    ): PurchaseWithDetailsOfferingContentTransformed => {
+      switch (planId) {
+        case 'plan1':
+          return {
+            puchaseDetails: {},
+          } as unknown as PurchaseWithDetailsOfferingContentTransformed;
+        case 'plan2':
+          return {
+            offering: { commonContent: {} },
+          } as unknown as PurchaseWithDetailsOfferingContentTransformed;
+        case 'plan5':
+          return PurchaseWithDetailsOfferingContentTransformedFactory({
+            purchaseDetails: PurchaseDetailsTransformedFactory({
+              webIcon: 'http://contentful.example/webicon',
+            }),
+          });
+        default:
+          return PurchaseWithDetailsOfferingContentTransformedFactory();
+      }
+    },
+  };
```

Comment:
It would be easier to read/understand if you had separate mock implementations for each scenario, and did the mock for each separately, rather than having a util that returns based on magic values

Reply:
Yup that makes sense. Let me get on refactoring this.